File: 3434.maximum-frequency-after-subarray-operation.py
#
# @lc app=leetcode id=3434 lang=python3
#
# [3434] Maximum Frequency After Subarray Operation
#

# @lc code=start
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        left = 0
        total = 0
        max_freq = 0
        # Using sliding window technique to find maximum frequency
        for right in range(len(nums)):
            total += nums[right]
            # Check if we can make all numbers in current window equal to nums[right]
            while (right - left + 1) * nums[right] - total > k:
                total -= nums[left]
                left += 1
            # Update max frequency
            max_freq = max(max_freq, right - left + 1)
        return max_freq
# @lc code=end

File: 3434.maximum-frequency-after-subarray-operation.cpp
#
# @lc app=leetcode id=3434 lang=cpp
#
# [3434] Maximum Frequency After Subarray Operation
#

# @lc code=start
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        long long totalOperations = 0;
        int left = 0;
        int maxFreq = 0;
        for (int right = 0; right < nums.size(); ++right) {
            totalOperations += nums[right];
            while ((right - left + 1) * nums[right] - totalOperations > k) {
                totalOperations -= nums[left];
                ++left;
            }
            maxFreq = max(maxFreq, right - left + 1);
        }
        return maxFreq;
    }
};
# @lc code=end